rir/src/compiler/opt/force_dominance.cpp

```diff
@@ -177,6 +190,20 @@ struct ForcedBy {
             }
         }
 
+        for (auto& mc : other.promToValueCast) {
+            if (promToValueCast.contains(mc.first)) {
```

Comment:
this is slow and causes analysis times to explode.

please use `p = promToValueCast.find(mc.first)`, then you can check with `p == promToValueCast.end()` and access with `p->second`